topology/metrics.go

```diff
@@ -232,3 +232,190 @@ func (im *InterfaceMetric) Split(cut int64) (common.Metric, common.Metric) {
 
 	return m1, m2
 }
+
+// BlockdevMetric the blockdevice IO stats
+// easyjson:json
+// gendecoder
+type BlockdevMetric struct {
+	ReadsPerSec        float64 `json:"r/s"`
+	WritesPerSec       float64 `json:"w/s"`
+	ReadsKBPerSec      float64 `json:"rkB/s"`
+	WritesKBPerSec     float64 `json:"wkB/s"`
+	ReadsMergedPerSec  float64 `json:"rrqm/s"`
+	WritesMergedPerSec float64 `json:"wrqm/s"`
+	ReadsMerged        float64 `json:"rrqm"`
+	WritesMerged       float64 `json:"wrqm"`
+	ReadServiceTime    float64 `json:"r_await"`
+	WriteServiceTime   float64 `json:"w_await"`
+	AverageQueueSize   float64 `json:"aqu-sz"`
+	AverageReadSize    float64 `json:"rareq-sz"`
+	AverageWriteSize   float64 `json:"wareq-sz"`
+	ServiceTime        float64 `json:"svctm"`
+	Utilization        float64 `json:"util"`
+	Start              int64   `json:"Start"`
+	Last               int64   `json:"Last"`
+}
+
+// GetBlockdevFields returns the JSON annotation for each field in
+// BlockdevMetric
+func GetBlockdevFields() []string {
+	return []string{"r/s",
+		"w/s",
+		"rkB/s",
+		"wkB/s",
+		"rrqm/s",
+		"wrqm/s",
+		"rrqm",
+		"wrqm",
+		"r_await",
+		"w_await",
+		"aqu-sz",
+		"rareq-sz",
+		"wareq-sz",
+		"svctm",
+		"util",
+		"Start",
+		"Last",
+	}
+}
+
+// IsZero returns true if all the values are equal to zero
+func (im *BlockdevMetric) IsZero() bool {
```

Comment:
If don't plan to use gremlin to do operation on metrics this function is useless

Reply:
Thanks - I removed it.